vm/src/stdlib/posix.rs

```diff
@@ -1975,4 +1979,24 @@ pub mod module {
         res.map_err(|err| err.into_pyexception(vm))?;
         Ok(vm.ctx.new_int(written as u64).into())
     }
+
+    #[cfg(target_os = "linux")]
+    #[pyfunction]
+    fn getrandom(size: isize, flags: OptionalArg<u32>, vm: &VirtualMachine) -> PyResult<Vec<u8>> {
+        let size = usize::try_from(size)
+            .map_err(|_| vm.new_os_error(format!("Invalid argument for size: {}", size)))?;
+        let mut buf = Vec::with_capacity(size);
+        let len = unsafe {
+            libc::getrandom(
+                buf.as_mut_ptr() as *mut libc::c_void,
+                buf.len(),
+                flags.unwrap_or(0),
+            )
+        };
+        if len == -1 {
+            return Err(errno_err(vm));
+        }
+        buf.set_len(len as usize);
```

Comment:
these lines also need to be placed in unsafe block

Reply:
Planed to use `shrink_to` which is safe and new (cool!)

Reply:
`buf.len()` here is `0`. `shrink_to` changes the capacity not the length.

Reply:
I end up using `truncate` which is safe and deals with dropping the rest of the elements.